pkg/sql/pgwire/testdata/pgtest/decimal

```diff
@@ -35,3 +35,36 @@ ReadyForQuery
 {"Type":"DataRow","Values":[{"text":"0"}]}
 {"Type":"CommandComplete","CommandTag":"SELECT 1"}
 {"Type":"ReadyForQuery","TxStatus":"I"}
+
+send
+Parse {"Name": "s1", "Query": "SELECT 10000::decimal, $1::decimal"}
+Bind {"DestinationPortal": "p1", "PreparedStatement": "s1", "ParameterFormatCodes": [1], "Parameters": [[0, 1, 0, 1, 0, 0, 0, 0, 0, 1]]}
+Execute {"Portal": "p1"}
+Sync
+----
+
+until
+ReadyForQuery
+----
+{"Type":"ParseComplete"}
+{"Type":"BindComplete"}
+{"Type":"DataRow","Values":[{"text":"10000"},{"text":"10000"}]}
```

Comment:
can you add one for 10E4 too?

Reply:
you mean add an expected return value of `10E4`? i don't think i understand -- postgres will never return a decimal value with that text format (and this is the `noncrdb_only` expected result)

is 10E4 what we would get if we add `0000` to the encoding above? sorry i don't understand!

Reply:
Ah I meant to parse 10E4

Reply:
done! i think that shows why CRDB uses scientific notation -- it lets us preserve the number of significant digits (as detailed in https://github.com/cockroachdb/cockroach/pull/17029)